jupyter_server/services/contents/fileidmanager.py

```diff
@@ -0,0 +1,126 @@
+import os
+import sqlite3
+
+from jupyter_core.paths import jupyter_data_dir
+from traitlets import Unicode
+from traitlets.config.configurable import LoggingConfigurable
+
+
+class FileIdManager(LoggingConfigurable):
+    db_path = Unicode(
+        default_value=os.path.join(jupyter_data_dir(), "file_id_manager.db"),
+        help=(
+            "The path of the DB file used by `FileIdManager`. "
+            "Defaults to `jupyter_data_dir()/file_id_manager.db`."
+        ),
+        config=True,
+    )
+
+    def __init__(self, *args, **kwargs):
+        # pass args and kwargs to parent Configurable
+        super().__init__(*args, **kwargs)
+        # initialize connection with db
+        self.con = sqlite3.connect(self.db_path)
```

Comment:
I believe adding `isolation_level=None` to the  `connect` method auto commits all changes to the DB, so you don't have to called `self.con.commit()` after every execute method.

For example, https://github.com/jupyter-server/jupyter_server/blob/e59610b6c7270e2987979b079b87cc9ef9d6ad2d/jupyter_server/services/sessions/sessionmanager.py#L216

Reply:
This severely degrades performance when doing a lot of writes in serial, which is why that is not the default. This happens in File ID manager when handling recursive moves/deletes/copies.

Why? Well, committing a transaction necessarily results in a file write, and there is also overhead with repeatedly translating your SQL statements to SQLite VDBE code. See here: https://stackoverflow.com/questions/1711631/improve-insert-per-second-performance-of-sqlite